### DISCUSSION TITLE
DHCP for W5500 weirdly slow.

### DISCUSSION THREAD

**Discussion by the-maldridge** [earlephilhower/arduino-pico#2165] [Q&A]
I've built out a board which has a Wiznet W5500 on it that provides an ethernet interface for when wireless operation just isn't the right choice, and have some code that handles switching between wired and wireless operation when the ethernet link comes up.  What does not work reliably, however, is that DHCP on ethernet seems to just hang and not really work.  I can see the Pico asking for a lease on the line with tcpdump and I can see my DHCP server offering a lease back, but for some reason the Pico doesn't pick up the lease.

This doesn't seem like the same issue from https://github.com/earlephilhower/arduino-pico/discussions/2149, so I've opened a new thread to seek advice.  The complete code I'm running is available [here](https://github.com/gizmo-platform/firmware/tree/enet2) with the interesting functions all being part of the `netStateMachine()`.  I'm using administrative MAC addresses, but I can't see how those could be a problem.  Any advice is appreciated.

**Comment by JAndrassy** [comment]
specify the INT pin in constructor (and wire it)

**Reply by the-maldridge** [reply]
This seems like an odd requirement when the lwIP polling interval is already down to 10ms, which should be plenty fast to catch the DHCP packets.  It also seems fishy that this would be required since using the Arduino Ethernet library instead of the vendored one from the toolchain has no issues at all pulling DHCP.

**Reply by the-maldridge** [reply]
As suspected, configuring and connecting the interrupt line does not make any observable change to the system.  Further investigation, however, has shown that using the vendored ethernet library DHCP never successfully completes.
